contracts/Market.sol

```diff
@@ -344,6 +349,34 @@ contract Market is IMarket, Ownable, ERC721 {
 		_burn(index);
 	}
 
+	function _applyScratchings(uint64 index) internal virtual returns (uint256) {
+		// Get marketId of bet
+		bytes16 marketId = _bets[index].marketId;
+		// Ask the oracle for scratched runners on this market
+		IOracle.Result memory result = IOracle(_oracle).getResult(marketId);
+
+		// Get all scratchings with a timestamp after this bet
+		// Loop through scratchings
+		uint256 scratchedOdds = 0;
+		uint256 betCreated = _bets[index].created;
+		for (uint256 i = 0; i < result.scratched.length; i++) {
```

Comment:
Same here with the .length.  Create a temp var to store length so the method costs less gas.